db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb

```diff
@@ -51,4 +33,38 @@ def down
                          .update_all(host_type: 'Hostgroup', host_id: hostgroup_facet_ids[hostgroup_facet_id])
     end
   end
+
+  private
+
+  def migrate_host_data_to_facets
+    host_config_group_ids = FakeHostConfigGroup.for_host.pluck(:host_id).uniq
+    host_config_group_ids.each do |host_id|
+      host_facet_id = ForemanPuppet::HostPuppetFacet.where(host_id: host_id).pick(:id)
+      host_groups = FakeHostConfigGroup.for_host.where(host_id: host_id)
+      if host_facet_id
+        host_groups.update_all(host_type: 'ForemanPuppet::HostPuppetFacet', host_id: host_facet_id)
+      else
+        deleted_groups = FakeConfigGroup.where(id: host_groups.pluck(:config_group_id)).pluck(:name)
+        say "deleting groups #{deleted_groups.join(', ')} from Host (id=#{host_id}) because it doesn't exists or has no environment"
+        host_groups.delete_all
+      end
+    end
+  end
+
+  def migrate_hostgroup_data_to_facets
+    hostgroup_config_group_ids = FakeHostConfigGroup.for_hostgroup.pluck(:host_id).uniq
+    hostgroup_config_group_ids.each do |hostgroup_id|
+      hostgroup = Hostgroup.unscoped.find_by(id: hostgroup_id)
+      hostgroup_groups = FakeHostConfigGroup.for_hostgroup.where(host_id: hostgroup_id)
```

Comment:
Why is it `host_id: hostgroup_id`?

Reply:
Because it's actually polymorphic association even though it's named host. I know confusing, but the more interesting part is, that for classes we have separate tables, so we does it differently for each model 🎉  but it's what we're working with coming from core. I'm in favor of changing it, but only after 2.0 is out :)